Applications/FileIO/CsvInterface.h

```diff
@@ -36,9 +39,36 @@ namespace FileIO {
 /**
  * Interface for reading CSV file formats.
  */
-class CsvInterface {
+class CsvInterface  : public BaseLib::IO::Writer
+{
 
 public:
+    /// Contructor (only needed for writing files)
+    CsvInterface() {};
+
+    /// Adds an index vector of size s to the CSV file
+    void addIndexVectorForWriting(std::size_t s);
+
+    /// Adds a data vector to the CSV file. All data vectors have to have the same size.
+    /// Vectors will be written in the same sequence they have been added to the interfaceW.
+    template<typename T>
+    bool addVectorForWriting(std::string const& vec_name, std::vector<T> const& vec)
+    {
+        if (typeid(vec) == typeid(std::vector<std::string>) ||
+            typeid(vec) == typeid(std::vector<double>) ||
+            typeid(vec) == typeid(std::vector<int>))
```

Comment:
I'm still not sure what you mean. The interface should be general so the programme shouldn't crash/exit if a vector of the wrong type is given.

Example: You load a bunch of data sets into the data explorer and decide you want to write all the parameters of *one* data set as a CSV file.  You cannot know at compile time which type of params the data set contains so you need to check at run-time. If a param is in there whose type is not supported, it should not crash the programme but instead be able to write all other params and give an error message that one param could not be written. 

Does that help?